dashboard/app/models/activity.rb

```diff
@@ -78,28 +78,23 @@ def Activity.recent(limit)
   # not pass validation. The object is only written asynchronously if the gatekeeper allows it for
   # this hostname.
   def Activity.create_async!(attributes)
+    activity = Activity.new(attributes)
+    activity.created_at = activity.updated_at = Time.now
+    activity.validate!
+    async_op = {'model' => 'Activity', 'action' => 'create', 'attributes' => activity.attributes}
```

Comment:
Note I improved this a bit. If async writes are on, we enqueue the async op in the queue, otherwise we execute the handler immediately.  This makes the code paths much closer in the two cases; the only difference is exactly when the handler is invoked.  The UserLevel writes follow the same pattern.